spiffetls/tests/check_mode.c

```diff
@@ -141,6 +147,95 @@ START_TEST(test_spiffetls_MTLSWebClientWithRawConfig)
 }
 END_TEST
 
+START_TEST(test_spiffetls_TLSServer)
+{
+    spiffetls_ListenMode *mode = spiffetls_TLSServer();
+
+    ck_assert_ptr_ne(mode, NULL);
+    ck_assert_uint_eq(mode->mode, TLS_SERVER_MODE);
+
+    spiffetls_ListenMode_Free(mode);
+}
+END_TEST
+
+START_TEST(test_spiffetls_TLSServerWithSource)
+{
+    err_t err;
+    workloadapi_X509Source *source = workloadapi_NewX509Source(NULL, &err);
+    ck_assert_uint_eq(err, NO_ERROR);
+    spiffetls_ListenMode *mode = spiffetls_TLSServerWithSource(source);
+
+    ck_assert_ptr_ne(mode, NULL);
+    ck_assert_uint_eq(mode->mode, TLS_SERVER_MODE);
+    ck_assert_ptr_eq(mode->source, source);
+
+    spiffetls_ListenMode_Free(mode);
+}
+END_TEST
+
+START_TEST(test_spiffetls_TLSServerWithRawConfig)
+{
+    x509svid_Source *svid = NULL;
```

Comment:
Shouldn't we use a non-null pointer for this? null pointers should give us an error